### PR TITLE
Added coverage test 0503

### PR DESCRIPTION
**Pull request by Vihanakangas**

### CODE REVIEW

File: coverage/PENDING_SUBMIT/intel/0503-COVERAGE-anv-nir-apply-pipeline-layout-560/cov-loop-condition-clamp-vec-of-ones.amber [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+# '--inline-entry-points-exhaustive'
+# '--scalar-replacement=100'
+# '--ccp'
+# spirv-opt commit hash: a0370efd589be33d5d9a85cfde2f85841b3755af
+
+
+
+SHADER vertex texgen_vert PASSTHROUGH
+
+SHADER fragment texgen_frag GLSL
```
[the remark this answers is not in the document]
Comment (asuonpaa) [reply coverage/PENDING_SUBMIT/intel/0503-COVERAGE-anv-nir-apply-pipeline-layout-560/cov-loop-condition-clamp-vec-of-ones.amber:57]: No it doesn't. An updated version gets rid of this.

**Comment by Vihanakangas** [comment]
@afd Updated.

File: coverage/PENDING_SUBMIT/intel/0503-COVERAGE-anv-nir-apply-pipeline-layout-560/cov-loop-condition-clamp-vec-of-ones.amber [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+#     float f = _float_1_0;
+#
+#     // Iterated once.
+#     do
+#     {
+#         // Iterated once.
+#         for (int i = 0; i != clamp(_int_0, int(one.x), 2); i++)
+#         {
+#             // Always chooses 2.0, thus f becomes 0.5.
+#             f /= one.y > _float_1_0 ? 256.0 : 2.0;
```
Comment (afd) [inline coverage/PENDING_SUBMIT/intel/0503-COVERAGE-anv-nir-apply-pipeline-layout-560/cov-loop-condition-clamp-vec-of-ones.amber:98]: Vulkan SPIR-V allows division to have some roundoff error:

https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#spirvenv-precision-operation

OpFDiv(x,y) is allowed 2.5 ULPs when |y| is in the range [2-126, 2126].

Thus f / one.y *could* end up being slightly larger than 1.0 if this round-off is used.

How about changing this to:

`f /= one.y > _float_1.0 + 0.1 ? 256.0 : 2.0`

?

Would that still get the coverage?

We did once have a SwiftShader "bug" that turned out not to be a bug due to a problem with dividing by 1.0., so I'm a little nervous.

**Review by afd — CHANGES_REQUESTED** [review]

File: coverage/PENDING_SUBMIT/intel/0503-COVERAGE-anv-nir-apply-pipeline-layout-560/cov-loop-condition-clamp-vec-of-ones.amber (hunk 2 above)
Comment (asuonpaa) [reply coverage/PENDING_SUBMIT/intel/0503-COVERAGE-anv-nir-apply-pipeline-layout-560/cov-loop-condition-clamp-vec-of-ones.amber:98]: I think you got the precedence wrong. The right hand side is calculated first so you can think of it as `f /= (one.y > _float_1_0 ? 256.0 : 2.0)`. So all the line does is f /= 2.0. We could add some threshold to the result checking though.

**Comment by Vihanakangas** [comment]
@afd Updated.

**Review by afd — APPROVED** [review]

**Pull request merged**